jwt.js

```diff
@@ -176,13 +222,15 @@ function fastifyJwt (fastify, options, next) {
   function replySign (payload, options, next) {
     if (typeof options === 'function') {
       next = options
-      options = Object.assign({}, signOptions)
+      options = {}
     } // support no options
 
     if (!options) {
-      options = Object.assign({}, signOptions)
+      options = {}
     }
 
+    options = Object.assign({}, signOptions, options.sign || {})
```

Comment:
nullable is fine `Object.assign({}, {a:1}, null)`.

I don't understand why you need this change.
The `options` object set by the user is always overwritten.

Right now, you can write:

```
reply. jwtSign({}, {issuer: 'foo'})
```

After this change you need to write:

```
reply. jwtSign({}, {sign: { issuer: 'foo'}})
```

Could you elaborate?

```suggestion
    options = Object.assign({}, signOptions, options.sign)
```